docs/src/model_setup/callbacks.md

```diff
@@ -21,30 +21,57 @@ show_time(sim) = @info "Time is $(prettytime(sim.model.clock.time))"
 
 simulation.callbacks[:total_A] = Callback(show_time, IterationInterval(2))
 
+simulation
+```
+
+Now when we run the simulation the callback is called.
+
+```@example checkpointing
 run!(simulation)
 ```
 
-State callbacks are useful for inter step modification of the model state (for example if you wanted to manually modify the tendency fields). Irrespective of the specified scheduling state callbacks are executed at every sub-step. As an example we can manually add to the tendency field of one of the velocity components, here I've chosen the `:u` field using parameters:
+We can also use the convenience [`add_callback!`](@ref). For example, we could add a callback like
 
-```@repl checkpointing
+```@example checkpointing
+add_callback!(simulation, show_time, name=:total_A_via_convenience, IterationInterval(2))
+
+simulation
+```
+
+By default, callbacks are called after a time-step has been completed but users
+can construct callbacks that are called after tendencies are calculated, but before taking a time-step.
+The latter is useful for modifying tendency calculations.
+
+We can control when the callback is called via the `callsite` keyword argument. By default
+`callsite = TimeStepCallsite()`. If we want the callback after the tendencies are
+computed but before taking a time-step we need to construct the callback with `callsite = TendencyCallsite()`.
```

Comment:
```suggestion
By default, callbacks are called after a time-step has been completed but users
can construct callbacks that are called after tendencies are calculated, but before taking a time-step.
The latter is useful for modifying tendency calculations.

We can control when the callback is called via the `callsite` keyword argument. By default
`callsite = TimeStepCallsite()`. If we want the callback after the tendencies are
computed but before taking a time-step we need to construct the callback with `callsite = TendencyCallsite()`.
```

These paragraphs repeat information which makes them hard to understand / ambiguous. Try to condense it. Something like

```
The keyword argument `callsite` determines the moment at which the callback is executed.
By default, `callsite = TimeStepCallsite()`, indicating execution _after_ the completion of a time-step.
Two other options are... [state options, and how they can be used].
```

Reply:
indeed! done